jmclient/jmclient/wallet.py

```diff
@@ -1334,7 +1343,7 @@ class LegacyWallet(ImportWalletMixin, BIP32Wallet):
     _ENGINE = BTC_P2PKH
 
     def _create_master_key(self):
-        return hexlify(self._entropy).encode('ascii')
+        return self._entropy
```

Comment:
so this should be `return hexlify(self._entropy)`? The `encode('ascii')` part is not valid for hexlify since hexlify returns bytes already. I tried `decode('ascii')` which also doesn't work due to this:
```python
test/test_segwit.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/common.py:99: in make_wallets
    test_wallet_cls=walletclass)
jmclient/jmclient/wallet_utils.py:902: in open_test_wallet_maybe
    return test_wallet_cls(storage, **kwargs)
jmclient/jmclient/wallet.py:814: in __init__
    super(ImportWalletMixin, self).__init__(storage, **kwargs)
jmclient/jmclient/wallet.py:1071: in __init__
    self._master_key = self._derive_bip32_master_key(_master_entropy)
jmclient/jmclient/wallet.py:1153: in _derive_bip32_master_key
    return cls._ENGINE.derive_bip32_master_key(seed)
jmclient/jmclient/cryptoengine.py:148: in derive_bip32_master_key
    btc.bip32_master_key(seed, vbytes=cls.BIP32_priv_vbytes))
jmbitcoin/jmbitcoin/secp256k1_deterministic.py:82: in bip32_master_key
    from_string_to_bytes("Bitcoin seed"), seed, hashlib.sha512).digest()
jmvenv/lib/python3.5/hmac.py:144: in new
    return HMAC(key, msg, digestmod)
jmvenv/lib/python3.5/hmac.py:84: in __init__
    self.update(msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <hmac.HMAC object at 0x7f2ae692c748>, msg = '3b9508177c69f563be16995b9c45e1f5'

    def update(self, msg):
        """Update this hashing object with the string msg.
        """
>       self.inner.update(msg)
E       TypeError: Unicode-objects must be encoded before hashing
```

Reply:
yes, in that case this needs to be `return hexlify(self._entropy)`.